packages/kit/src/build.ts

```diff
@@ -1,135 +1,168 @@
 
 /**
- * Extend webpack config
- *
- * The fallback function might be called multiple times
- * when applying to both client and server builds.
+ * Extends the webpack configuration. Callback function can be called multiple times, when applying to both client and server builds.
+ * @param callbackFunction - A callback function that will be called with the webpack configuration object.
+ * @param options - Options to pass to the callback function. See {@link https://nuxt.com/docs/api/kit/builder#options available options}.
+ * @see {@link https://nuxt.com/docs/api/kit/builder#extendwebpackconfig documentation}
  */
 export function extendWebpackConfig (
-  fn: ((config: WebpackConfig) => void),
+  callbackFunction: ((config: WebpackConfig) => void),
   options: ExtendWebpackConfigOptions = {}
 ) {
   const nuxt = useNuxt()
 
-  if (options.dev === false && nuxt.options.dev) {
-    return
-  }
-  if (options.build === false && nuxt.options.build) {
+  if (
+    (options.dev === false && nuxt.options.dev)
+    || (options.build === false && !nuxt.options.dev)
+  ) {
     return
   }
 
   nuxt.hook('webpack:config', (configs: WebpackConfig[]) => {
     if (options.server !== false) {
-      const config = configs.find(i => i.name === 'server')
+      const config = configs.find((index) => index.name === 'server')
+
       if (config) {
-        fn(config)
+        callbackFunction(config)
       }
-    }
-    if (options.client !== false) {
-      const config = configs.find(i => i.name === 'client')
+    } else if (options.client !== false) {
+      const config = configs.find((index) => index.name === 'client')
+
       if (config) {
-        fn(config)
+        callbackFunction(config)
       }
     }
   })
 }
 
 /**
- * Extend Vite config
+ * Extends the Vite configuration. Callback function can be called multiple times, when applying to both client and server builds.
+ * @param callbackFunction - A callback function that will be called with the Vite configuration object.
+ * @param options - Options to pass to the callback function. See {@link https://nuxt.com/docs/api/kit/builder#options-1 available options}.
+ * @see {@link https://nuxt.com/docs/api/kit/builder#extendviteconfig documentation}
  */
 export function extendViteConfig (
-  fn: ((config: ViteConfig) => void),
+  callbackFunction: ((config: ViteConfig) => void),
   options: ExtendViteConfigOptions = {}
 ) {
   const nuxt = useNuxt()
 
-  if (options.dev === false && nuxt.options.dev) {
-    return
-  }
-  if (options.build === false && nuxt.options.build) {
+  if (
+    (options.dev === false && nuxt.options.dev)
+    || (options.build === false && !nuxt.options.dev)
+  ) {
     return
   }
 
   if (options.server !== false && options.client !== false) {
-    // Call fn() only once
-    return nuxt.hook('vite:extend', ({ config }) => fn(config))
+    // Call callbackFunction() only once
+    nuxt.hook('vite:extend', ({ config }) => {
+      callbackFunction(config)
+    })
```

Comment:
It's important the promise is awaited.

```suggestion
    nuxt.hook('vite:extend', ({ config }) => callbackFunction(config))
```